pinot-spi/src/main/java/org/apache/pinot/spi/utils/CommonConstants.java

```diff
@@ -657,6 +663,10 @@ public static class QueryOptionKey {
         // When lite mode is enabled, if this flag is set, we will run all the non-leaf stage operators within the
         // broker itself. That way, the MSE queries will model the scatter gather pattern used by the V1 Engine.
         public static final String RUN_IN_BROKER = "runInBroker";
+
+        // For MSE queries, when this option is set to true, only use servers for leaf stages as the workers for the
+        // intermediate stages. This is useful to control the fanout of the query and reduce data shuffling.
+        public static final String USE_LEAF_SERVER_FOR_INTERMEDIATE_STAGE = "useLeafServerForIntermediateStage";
```

Comment:
nit: can we use the three `///` to be recognized as Javadoc? I know some other constants are not documented, but new ones should